### PR TITLE
Add AnimeKO

### PR DESCRIPTION
**Pull request by akumachi**
Request to add website AnimeKO

AnimeKO is a french streaming website.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
   "version": "0.9.3",
   "description": "Integrates MyAnimeList/AniList/Kitsu/Simkl into various sites, with auto episode tracking.",
   "engines": {
-    "npm": "^8.0.0",
```
Comment (lolamtisch) [inline package.json:10]: Please dont commit package.json and package-lock.json changes

File: src/pages/AnimeKO/main.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      return utils.absoluteLink(j.$('.small-card h2 a').attr('href'), AnimeKO.domain);
+    },
+    getEpisode(url) {
+      return Number(j.$('.player-releases :selected').text().split(' ')[1]);
+    },
+    nextEpUrl(url) {
+      return utils.absoluteLink(j.$('.btn-next').attr('href'), AnimeKO.domain);
+    },
+    getMalUrl(provider) {
+      return 'https://myanimelist.net/anime/' + jsonData.mal_id;
```
Comment (lolamtisch) [inline src/pages/AnimeKO/main.ts:36]: Dangerous. An empty mal_id would result in a broken myanimelist url. Please check that not empty.

File: src/pages/AnimeKO/main.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  },
+  overview: {
+    getTitle(url) {
+      return j.$('.showcase h1').text().trim();
+    },
+    getIdentifier(url) {
+      return utils.urlPart(url, 4);
+    },
+    getMalUrl() {
+      return 'https://myanimelist.net/anime/' + jsonData.mal_id;
```
Comment (lolamtisch) [inline src/pages/AnimeKO/main.ts:47]: Same as above

File: src/pages/AnimeKO/main.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      api.storage.addStyle(
+        require('!to-string-loader!css-loader!less-loader!./style.less').toString(),
+      );
+      j.$(function () {
+        if (j.$('#malsync').length) {
+          jsonData = JSON.parse(j.$('#malsync').text());
+        }
+      });
+      page.handlePage();
+      utils.urlChangeDetect(function () {
```
Comment (lolamtisch) [inline src/pages/AnimeKO/main.ts:76]: As far as I see does the page always reload/new page, so listening for url changes is not necessary?

File: src/pages/AnimeKO/main.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        return utils.absoluteLink(selector.find('a').attr('href'), AnimeKO.domain);
+      },
+    },
+  },
+  init(page) {
+    j.$(function () {
+      api.storage.addStyle(
+        require('!to-string-loader!css-loader!less-loader!./style.less').toString(),
+      );
+      j.$(function () {
```
Comment (lolamtisch) [inline src/pages/AnimeKO/main.ts:70]: Why the seperate function here?

File: src/pages/AnimeKO/main.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  },
+  isSyncPage() {
+    return !!j.$('.player-wrap').length;
+  },
+  sync: {
+    getTitle(url) {
+      return j.$('.small-card h2').text().trim();
+    },
+    getIdentifier(url) {
+      return utils
```
Comment (lolamtisch) [inline src/pages/AnimeKO/main.ts:21]: Here it is probably better to use `return AnimeKO.sync.getIdentifier(AnimeKO.overview!.getOverviewUrl(url))` to follow DRY.

File: src/pages/AnimeKO/main.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        .pop();
+    },
+    getOverviewUrl(url) {
+      return utils.absoluteLink(j.$('.small-card h2 a').attr('href'), AnimeKO.domain);
+    },
+    getEpisode(url) {
+      return Number(j.$('.player-releases :selected').text().split(' ')[1]);
+    },
+    nextEpUrl(url) {
+      return utils.absoluteLink(j.$('.btn-next').attr('href'), AnimeKO.domain);
```
Comment (lolamtisch) [inline src/pages/AnimeKO/main.ts:33]: check if .btn-next exists

**Review by lolamtisch — CHANGES_REQUESTED** [review]

File: src/pages/AnimeKO/main.ts (hunk 6 above)
Comment (akumachi) [reply src/pages/AnimeKO/main.ts:21]: this throws an error of "Property 'getOverviewUrl' does not exist"

File: .gitignore [hunk 8]
```diff
@@ -45,3 +45,8 @@ bwd/
 dist/
 
 src/utils/quicklinks.json
+
+# IDE
+.idea
+package.json
+package-lock.json
```
Comment (lolamtisch) [inline .gitignore:52]: This is not the way to remove the package.json changes. You will need to revert or cherrypick.

File: src/pages/AnimeKO/main.ts [hunk 9]
```diff
@@ -0,0 +1,85 @@
+import { pageInterface } from '../pageInterface';
+
+let jsonData;
+
+export const AnimeKO: pageInterface = {
+  name: 'AnimeKO',
+  domain: 'https://animeko.co/',
```
Comment (lolamtisch) [inline src/pages/AnimeKO/main.ts:7]: Please remove the / in the end

File: src/pages/AnimeKO/main.ts (hunk 6 above)
Comment (lolamtisch) [reply src/pages/AnimeKO/main.ts:21]: Was the other way around
`return AnimeKO.overview!.getIdentifier(AnimeKO.sync.getOverviewUrl(url))`

File: src/pages/AnimeKO/style.less [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  background-color: white !important;
+}
+
+.floatbutton {
+  right: 20px;
+  bottom: 80px;
+}
+
+.mal-sync-active {
+  background-color: #4A148C;
```
Comment (lolamtisch) [inline src/pages/AnimeKO/style.less:23]: Please run `npm run lint:fix`

File: src/pages/AnimeKO/tests.json [hunk 11]
```diff
@@ -0,0 +1,28 @@
+{
+  "title": "AnimeKO",
+  "url": "https://animeko.co/",
+  "testCases": [
+    {
+      "url": "http://animeko.co/skip-to-loafer-episode-01-vostfr",
```
Comment (lolamtisch) [inline src/pages/AnimeKO/tests.json:6]: Here everything should probably be https://

File: src/pages/AnimeKO/main.ts [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+      return utils.urlPart(url, 4);
+    },
+    getMalUrl() {
+      if (jsonData.mal_id) {
+        return 'https://myanimelist.net/anime/' + jsonData.mal_id;
+      }
+      return false;
+    },
+    uiSelector(selector) {
+      j.$('.showcase');
```
Comment (lolamtisch) [inline src/pages/AnimeKO/main.ts:56]: This does not inject the UI. It is a labeled wrong. Selector is the html that needs to be injected
EX.: https://github.com/MALSync/MALSync/blob/master/src/pages/Animeflv/main.ts#L38

**Review by lolamtisch — CHANGES_REQUESTED** [review]

File: src/pages/AnimeKO/style.less [hunk 13]
```diff
@@ -0,0 +1,25 @@
+@import './../pages';
+
```
Comment (lolamtisch) [inline src/pages/AnimeKO/style.less:2]: The episode highlighting is quite ugly

image

File: src/pages/AnimeKO/style.less [hunk 14]
```diff
@@ -0,0 +1,25 @@
+@import './../pages';
```
Comment (lolamtisch) [inline src/pages/AnimeKO/style.less:1]: The UI styling is broken. Black text on black background
image

File: src/pages/AnimeKO/style.less (hunk 10 above)
Comment (lolamtisch) [inline src/pages/AnimeKO/style.less:23]: Has to be lowercase or the linter complains. `#4a148c`

**Review by lolamtisch — CHANGES_REQUESTED** [review]

**Pull request merged**